pkg/cmd/cli/cmd/projects.go

```diff
@@ -0,0 +1,175 @@
+	"github.com/openshift/origin/pkg/project/api"
+
+	"github.com/spf13/cobra"
+)
+
+type ProjectsOptions struct {
+	Config       clientcmdapi.Config
+	ClientConfig *restclient.Config
+	Client       *client.Client
+	Out          io.Writer
+	PathOptions  *kubecmdconfig.PathOptions
+
+	DisplayShort bool
+}
+
+const (
+	projectsLong = `
+Display information about the current active project and existing projects on the server.
+
+For advanced configuration, or to manage the contents of your config file, use the 'config'
+command.`
+
+	projectsExample = `  # Display the projects that currently exist
+  %[1]s`
+)
+
+// NewCmdProjects implements the OpenShift cli rollback command
+func NewCmdProjects(fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	options := &ProjectsOptions{}
+
+	cmd := &cobra.Command{
+		Use:     "projects",
+		Short:   "Display existing projects",
+		Long:    projectsLong,
+		Example: fmt.Sprintf(projectsExample, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			options.PathOptions = cliconfig.NewPathOptions(cmd)
+
+			if err := options.Complete(f, args, out); err != nil {
+				kcmdutil.CheckErr(kcmdutil.UsageError(cmd, err.Error()))
+			}
+
+			if err := options.RunProjects(); err != nil {
+				kcmdutil.CheckErr(err)
+			}
+		},
+	}
+
+	cmd.Flags().BoolVarP(&options.DisplayShort, "short", "q", false, "If true, display only the project names")
+	return cmd
+}
+
+func (o *ProjectsOptions) Complete(f *clientcmd.Factory, args []string, out io.Writer) error {
+	if len(args) > 0 {
+		return fmt.Errorf("no arguments should be passed")
+	}
+
+	var err error
+	o.Config, err = f.OpenShiftClientConfig.RawConfig()
+	if err != nil {
+		return err
+	}
+
+	o.ClientConfig, err = f.OpenShiftClientConfig.ClientConfig()
+	if err != nil {
+		return err
+	}
+
+	o.Client, _, err = f.Clients()
+	if err != nil {
+		return err
+	}
+
+	o.Out = out
+
+	return nil
+}
+
+// RunProjects lists all projects a user belongs to
+func (o ProjectsOptions) RunProjects() error {
+	config := o.Config
+	clientCfg := o.ClientConfig
+	out := o.Out
+
+	currentContext := config.Contexts[config.CurrentContext]
+	currentProject := currentContext.Namespace
+
+	var currentProjectExists bool = false
+	var currentProjectErr error = nil
+
+	client := o.Client
+
+	if len(currentProject) > 0 {
+		if _, currentProjectErr := client.Projects().Get(currentProject); currentProjectErr == nil {
+			currentProjectExists = true
+		}
+	}
+
+	defaultContextName := cliconfig.GetContextNickname(currentContext.Namespace, currentContext.Cluster, currentContext.AuthInfo)
+
+	var msg string
+	var current string
+	projects, err := getProjects(client)
+	if err == nil {
+		switch len(projects) {
+		case 0:
+			msg += "You are not a member of any projects. You can request a project to be created with the 'new-project' command."
+		case 1:
+			msg += fmt.Sprintf("\nYou have one project on this server: %s", api.DisplayNameAndNameForProject(&projects[0]))
```

Comment:
I think `\n` is not needed here, it would add a linebreak where other cases don't have it. Also, use `%q` instead of `%s` so that you have double quotes in the project name, and add a "." after it at the end of the phrase.